packages/as-transformer/src/helpers/protobuf.ts

```diff
@@ -88,18 +100,44 @@ ${response}
  *
  * @returns the protobuf argument as a string.
  */
-function generateArgumentMessage(arg: Argument, index: number): string {
+function generateArgumentMessage(
+  arg: Argument,
+  index: number,
+  transformer: MassaExport,
+): string {
   const fieldName = arg.name;
   const fieldSpec = getTypeName(arg.type);
-  const fieldType = (fieldSpec.repeated ? 'repeated ' : '') + fieldSpec.type;
-  return `  ${fieldType} ${fieldName} = ${index};`;
+  const typeName =
+    fieldSpec.type !== null && fieldSpec.type !== undefined
```

Comment:
The difference between type and cType is not clear, but if cType is customozed type, should'nt you first look for the presence of the optional customozed type instead of the ever existing protobuf type?

Reply:
Take it for what it worth: I'd suggest to leave it like this for now and plan to remove/merge `enum ProtoType` `getTypeName()` and `MassaCustomType` in possibly a single global map:

(pseudo code)
```ts
type ASType = string;
export interface MassaCustomType {
  name: string;
  meta_data?: TypeMetaData;
}

class TypeMetaData {
  proto: string;
  import: string;
  serialize: string;
  deserialize: string;
}

const GlobalTypeMap: Map<ASType, MassaCustomType>
```

for primitive types, the map should be initialized with what's in `enum ProtoType` `getTypeName()` setting no meta_data
when finding some types defined in a `.yml` append it to the map with meta_data for serialize / deserialize etc

I have some prepare work already done in my branch

Reply:
Follow up issue created, I take the :+1: for a "you can close this point"

Reply:
Yes, sorry. I have accepted the review as this is not a blocking point.

Reply:
:+1: (I did not check and thought that all conversations have to be closed for the PR to be able to merge, even if accepted)